fulcrumapp/tokml
# FLCRM-15076 update styles

This PR changes how <Style> nodes are generated. Instead of creating styles with only an <IconStyle>, or <LineStyle> and <PolyStyle>, all three styles are included for every styleHash. This provides better color display for all geometry types, and should make KML output by tools like Fulcrum, more reflective of simplestyle-spec attributes.

Relates to Fulcrum issue [FLCRM-15076](https://fulcrumapp.atlassian.net/browse/FLCRM-15076)

## Testing in Fulcrum
- Create a Form with Records that have a variety of Status colors
- Create a Shared View of the Records
- Use the "KML" tab to obtain the URL of the share

Note that putting the url into your browser, then renaming the file to have a `.kml` extension works better in most tools than using the URL directly. Google Earth only accepts files, while ArcGIS MapViewer only accepts a URL. 

Here are examples from the preview environment:

### Shared View in Fulcrum
![Screenshot 2024-05-08 at 12 40 32 PM](https://github.com/fulcrumapp/tokml/assets/2982313/39d0d4b0-8fa4-4370-a708-e3a4e681411c)

### KML output from Fulcrum
[test.kml.zip](https://github.com/fulcrumapp/tokml/files/15254041/test.kml.zip)

### [Google Earth](https://earth.google.com/web)
![Screenshot 2024-05-08 at 12 49 06 PM](https://github.com/fulcrumapp/tokml/assets/2982313/12c1f947-6ab8-4f9e-b437-de0e14be6083)

### QGIS
Note the icons are not rendered, but there are small shapes placed where each Point is located
![Screenshot 2024-05-08 at 12 55 07 PM](https://github.com/fulcrumapp/tokml/assets/2982313/365eb046-4adf-4c56-b45c-ad23c29a1bbc)

### [ArcGIS MapViewer](https://www.arcgis.com/apps/mapviewer/index.html)
Note that the `iconUrl`, taken from http://kml4earth.appspot.com/icons.html, does not get colored correctly, but passing a different url is supported if a better icon is found
![Screenshot 2024-05-08 at 12 59 42 PM](https://github.com/fulcrumapp/tokml/assets/2982313/d3bc441d-1f31-43bb-8598-b78c29e13aee)


[FLCRM-15076]: https://fulcrumapp.atlassian.net/browse/FLCRM-15076?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
